scapy/layers/dhcp6.py

```diff
@@ -1026,6 +1029,16 @@ class DHCP6OptClientLinkLayerAddr(_DHCP6OptGuessPayload):  # RFC6939
                    _LLAddrField("clladdr", ETHER_ANY)]
 
 
+class DHCP6OptMudUrl(_DHCP6OptGuessPayload):  # RFC8520
+    name = "DHCP6 Option - MUD URL"
+    fields_desc = [ShortEnumField("optcode", 112, dhcp6opts),
+                   FieldLenField("optlen", None, length_of="mudString"),
+                   StrLenField("mudString", "",
+                               length_from=lambda pkt: pkt.optlen,
+                               max_length=255,
```

Comment:
The issues should be fixed now :) 

Reading through the [specification](https://tools.ietf.org/html/rfc8520#section-10) once more, however, I stumpled upon the line

> The entire option MUST NOT exceed 255 octets. 

As the `optcode` and the `optlen` use up one byte each, does does this imply that the actual `max_length` of the mudString` should rather be 253 than 255?

Reply:
Since `optlen` can go up to 255, I'd say it's up to interpretation. You can chose 253 just to be safe, `max_length` is only used when random values are asked.

Reply:
Alright, thank you for your feedback on this :) I changed the value to 253 for the sake of accuracy.